bundles/org.openhab.binding.mercedesme/src/main/java/org/openhab/binding/mercedesme/internal/AccountHandler.java

```diff
@@ -0,0 +1,217 @@
+ */
+@NonNullByDefault
+public class AccountHandler extends BaseBridgeHandler implements AccessTokenRefreshListener {
+    private final Logger logger = LoggerFactory.getLogger(AccountHandler.class);
+    private final OAuthFactory oAuthFactory;
+    private final Storage<String> storage;
+    private final HttpClient httpClient;
+    private final MercedesMeTranslationProvider i18nProvider;
+    private Optional<CallbackServer> server = Optional.empty();
+    private Optional<String> tokenStorageKey = Optional.empty();
+
+    Optional<AccountConfiguration> config = Optional.empty();
+
+    public AccountHandler(Bridge bridge, HttpClient hc, OAuthFactory oaf, Storage<String> storage,
+            MercedesMeTranslationProvider translationProvider) {
+        super(bridge);
+        httpClient = hc;
+        oAuthFactory = oaf;
+        i18nProvider = translationProvider;
+        this.storage = storage;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // no commands available
+    }
+
+    @Override
+    public void initialize() {
+        config = Optional.of(getConfigAs(AccountConfiguration.class));
+        handleConfig();
+        String configValidReason = configValid();
+        if (!configValidReason.equals(Constants.EMPTY)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, configValidReason);
+        } else {
+            String callbackUrl = Utils.getCallbackAddress(config.get().callbackIP, config.get().callbackPort);
+            thing.setProperty("callbackUrl", callbackUrl);
+
+            server = Optional.of(new CallbackServer(this, httpClient, oAuthFactory, config.get(), callbackUrl));
+            tokenStorageKey = Optional.of(config.get().clientId + ":token");
+            if (storage.containsKey(tokenStorageKey.get())) {
+                String tokenSerial = storage.get(tokenStorageKey.get());
+                if (tokenSerial != null) {
+                    AccessTokenResponse atr = (AccessTokenResponse) Utils.fromString(tokenSerial);
+                    server.get().setToken(atr);
+                } else {
+                    logger.debug("Token cannot be restored from storage - manual authorization needed");
+                }
+            } else {
+                if (!keyMigration()) {
+                    logger.debug("Token not found in storage - manual authorization needed");
+                }
+            }
+
+            server.get().start();
+            String token = server.get().getToken();
+            if (!token.equals(Constants.EMPTY)) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                String textKey = MercedesMeTranslationProvider.PREFIX + thing.getThingTypeUID().getId()
+                        + MercedesMeTranslationProvider.STATUS_AUTH_NEEDED;
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, i18nProvider.getText(textKey) + callbackUrl);
```

Comment:
This can be simplified, since `updateStatus()` directly supports i18n text references - see https://github.com/openhab/openhab-addons/pull/13016#discussion_r937907377.

Reply:
Right, works like a charme